Library/Homebrew/extend/pathname.rb

```diff
@@ -289,14 +289,14 @@ def write_exec_script(*targets)
   end
 
   # Writes an exec script that sets environment variables.
-  sig { params(target: Pathname, args: T.any(T::Array[String], T::Hash[String, String]), env: T.nilable(T::Hash[String, String])).void }
+  sig { params(target: Pathname, args: T.any(String, T::Array[String], T::Hash[String, String]), env: T.nilable(T::Hash[String, String])).void }
   def write_env_script(target, args, env = nil)
     unless env
       env = args
       args = nil
     end
     env_export = +""
-    env.each { |key, value| env_export << "#{key}=\"#{value}\" " }
+    Array(env).each { |key, value| env_export << "#{key}=\"#{value}\" " }
     dirname.mkpath
     write <<~SH
```

Comment:
The handling of `args` below also needs to be fixed for when `args` is a `T::Array[String]`.